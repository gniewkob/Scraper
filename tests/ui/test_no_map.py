import subprocess
from pathlib import Path


def render_app() -> str:
    frontend_dir = Path(__file__).resolve().parents[2] / "frontend"
    script = (
        "import React from 'react';"
        "globalThis.React = React;"
        "import ReactDOMServer from 'react-dom/server';"
        "import App from './src/App.tsx';"
        "console.log(ReactDOMServer.renderToString(React.createElement(App)));"
    )
    result = subprocess.run(
        ["npx", "--yes", "tsx", "-e", script],
        capture_output=True,
        text=True,
        check=True,
        cwd=frontend_dir,
    )
    return result.stdout.strip()


def test_no_map():
    html = render_app()
    assert 'id="map"' not in html
    snapshot_file = Path(__file__).with_suffix('.snap')
    if snapshot_file.exists():
        assert html == snapshot_file.read_text()
    else:
        snapshot_file.write_text(html)
