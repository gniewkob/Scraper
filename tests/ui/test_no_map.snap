<div class="container py-4"><div class="text-end"><button id="themeToggle" class="btn btn-outline-light btn-sm">Zmień motyw</button></div><h1 class="text-center mb-4">🌿 Dashboard cen medycznej marihuany</h1><div class="row mb-3"><div class="col-md-6 mb-2 mb-md-0"><label class="form-label">Wybierz produkt:</label><select class="form-select"></select></div><div class="col-md-6"><label class="form-label">Miasto:</label><select class="form-select"><option value="" selected="">Wszystkie miasta...</option></select></div></div><div class="d-flex gap-2 my-2"><select class="form-select form-select-sm"><option value="price" selected="">Cena</option><option value="expiration">Ważność</option><option value="fetched_at">Data pobrania</option></select><select class="form-select form-select-sm"><option value="asc" selected="">Rosnąco</option><option value="desc">Malejąco</option></select></div><div class="card p-3 mb-4"><h2 class="card-title">💎 Najtańsze oferty</h2><div class="table-responsive"><table class="table table-bordered"><thead><tr><th>Cena (za 1 g)</th><th>Apteka</th><th>Adres</th><th>Mapa</th></tr></thead><tbody></tbody></table></div><div id="countInfo" class="mb-2 text-muted">Wyświetlasz <!-- -->1<!-- -->–<!-- -->0<!-- --> z <!-- -->0<!-- --> ofert</div></div><div class="card p-3 mb-4"><h2 class="card-title">📈 Trend cen wg daty</h2><canvas height="300"></canvas></div></div>